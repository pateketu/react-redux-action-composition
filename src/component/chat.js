import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import loadChat from '../redux/modules/chatActions';
import {formatDate} from '../utils/dateUtil';
export class Chat extends React.Component{   

    componentDidMount(){              
        this.props.loadChat();        
    }
    
    render(){                              
        return <div>
            {
                this.props.chat.map((chatItem) =>{
                    return <div key={chatItem.id} className='chatItem'>
                        {chatItem.message} by <b>{chatItem.user.firstName}</b>  @  {formatDate(chatItem.timestamp)}
                    </div>
                })
            }
        </div>;
    }
}

Chat.propTypes = {
    chat:PropTypes.array,
    loadChat:PropTypes.func.isRequired
}

export default connect(state => ({chat:state.chat}),
                        dispatch => (
                                {
                                    loadChat:()=>{dispatch(loadChat())
                                }
                        }))(Chat);