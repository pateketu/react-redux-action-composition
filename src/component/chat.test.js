import { Chat} from './chat';
import React from 'react';
import renderer from 'react-test-renderer';
describe('Chat Component', ()=>{
    it('Renders correctly', ()=>{
        const chat = [{
            "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
            "userId": "e837c9f5-247f-445f-bcc3-7d434348336b",
            "message": "Test Message",
            "timestamp": "2017-02-09T04:27:38Z",
            "user":{
                "id": "e837c9f5-247f-445f-bcc3-7d434348336b",
                "firstName": "Martin",
                "lastName": "Bradley",
                "email": "mbradley0@google.it",
                "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
                "ip": "166.124.172.160"
              }
          }];
        const tree = renderer.create(<Chat chat={chat} loadChat={()=>{}}></Chat>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});