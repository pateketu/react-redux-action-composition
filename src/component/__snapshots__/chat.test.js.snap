// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component Renders correctly 1`] = `
<div>
  <div
    className="chatItem"
  >
    Test Message
     by 
    <b>
      Martin
    </b>
      @  
    9/2/2017 4:27:38
  </div>
</div>
`;
